.githooks/pre-commit

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+echo "*** Running pre-commit checks ***"
+
+has_errors=0
+
+printf "Checking 'go mod tidy'...\r"
+cp go.mod go.mod.bak && cp go.sum go.sum.bak  # Backup the current go.mod and go.sum
+go mod tidy
+diff go.mod go.mod.bak > /dev/null
+mod_diff=$?
+diff go.sum go.sum.bak > /dev/null
+sum_diff=$?
+
+# Clean up backup files
+rm go.mod.bak go.sum.bak
+
+if [[ $mod_diff -ne 0 ]] || [[ $sum_diff -ne 0 ]];
+then
+    printf "Checking 'go mod tidy'... ❌\n"
+    printf "'go mod tidy' has not been run on your changes.\n"
+    printf "Run `go mod tidy` and commit the changes.\n"
+    exit 1
+else
+    printf "Checking 'go mod tidy'... ✅\n"
+fi
+
+printf "Checking go.mod for local replacements...\r"
+
+if grep -q "replace .* => /" "./go.mod"; then
+    has_errors=1
+    printf "Checking go.mod for local replacements... ❌\n"
+    grep -n "replace .* => /" "./go.mod"
```

Comment:
This only checks for local replaces that use absolute paths. Is this common? I may just do things weird but my local replaces are usually using relative paths so would start with ../ and wouldn't get caught by this, but if we did that here then we would block things like sub projects that use the main project as a relative local replace.